netsim/extra/bonding/plugin.py

```diff
@@ -0,0 +1,90 @@
+import typing
+from box import Box
+from netsim.utils import log
+from netsim import api,data
+from netsim.augment import devices
+
+_config_name = 'bonding'
+
+'''
+add_bond_interfaces - append interface data to node.interfaces for bonding template to read and implement
+'''
+def add_bond_interfaces(node: Box, bonds: dict[int,Box]) -> None:
+  for c,(ifindex,bond) in enumerate(bonds.items()):
+    ifname = f'bond{ifindex}'  # XXX hardcoded, could make this a device template attribute
+
+    #
+    # TODO: Could make sure the layout is exactly like the 'lag' module needs it, to reuse provisioning logic
+    # That means converting 'bonding.ifindex' to 'lag._parentindex' too!
+    #
+    bond_if = {
+      'type': 'bond',
+      'ifname': ifname,
+      'name': f'bond {ifindex}',
+      'bonding': { 'ifindex': ifindex, 'members': bond['members'], 'mode': bond.mode },
+      'interfaces': bond['interfaces'],
+      'ifindex': 50000 + c,
+      'virtual_interface': True
+    }
+    if 'primary' in bond:
+      bond_if['bonding']['primary'] = bond['primary']
+    for af in ['ipv4','ipv6']:
+      if af in bond:
+        bond_if[af] = bond[af]     # Take the first one, if any
+    node.interfaces.append(bond_if)
+
+'''
+post_transform hook
+
+Apply plugin config to nodes with interfaces marked with 'bonding.ifindex', for devices that support this plugin
+'''
+def post_transform(topology: Box) -> None:
+# def post_link_transform(topology: Box) -> None: # Use 'pre_node_transform' such that only 1 IP address gets assigned, nothing to move?
+  global _config_name
+  bond_mode = topology.get('bonding.mode','active-backup')
+  bonds : Box = data.get_empty_box()                   # Map of bonds per node, indexed by bonding.ifindex
+  for node in topology.nodes.values():
+    features = devices.get_device_features(node,topology.defaults)
```

Comment:
This seems to work along the lines of "ignore bonding attributes on nodes that do not support bonding". Not exactly the best approach to the Principle of Least Astonishment.

Reply:
I forgot that plugins don't automatically get these checks, unlike modules